src/mini_trainer/osft_utils.py

```diff
@@ -956,6 +1059,19 @@ def prepare_state_dict_for_save(self, state_dict):
                     upcast_dtype=self.upcast_dtype,
                 )
                 state_dict[orig] = W
+                
+                # Explicitly delete intermediate tensors to free memory
+                del U_high, S_high, V_high, U_low, S_low, V_low
```

Comment:
@NikhilNayak-debug This doesn't delete the tensors altogether? If we are popping them from the state dict then it would be a pass by reference, have you tested that these are copies and deleting them here doesn't also delete them from the state dict?